src/adapters/interfaces/IMorphoMarketV1Adapter.sol

```diff
@@ -3,31 +3,53 @@
 pragma solidity >=0.5.0;
 
 import {IAdapter} from "../../interfaces/IAdapter.sol";
-import {Id, MarketParams} from "../../../lib/morpho-blue/src/interfaces/IMorpho.sol";
+import {MarketParams} from "../../../lib/morpho-blue/src/interfaces/IMorpho.sol";
+
+struct MarketPosition {
+    uint128 supplyShares;
+    uint128 allocation;
+}
 
 interface IMorphoMarketV1Adapter is IAdapter {
     /* EVENTS */
 
+    event Allocate(bytes32 indexed marketId, uint256 newAllocation, uint256 shares);
+    event Deallocate(bytes32 indexed marketId, uint256 newAllocation, uint256 shares);
     event SetSkimRecipient(address indexed newSkimRecipient);
     event Skim(address indexed token, uint256 assets);
+    event SubmitBurnShares(bytes32 indexed id, uint256 executableAt);
+    event RevokeBurnShares(bytes32 indexed id);
+    event BurnShares(bytes32 indexed id, uint256 supplyShares);
 
     /* ERRORS */
 
+    error AlreadyPending();
+    error IrmMismatch();
     error LoanAssetMismatch();
     error NotAuthorized();
+    error NotPending();
+    error NotTimelocked();
+    error SharePriceAboveOne();
+    error TimelockNotExpired();
 
     /* FUNCTIONS */
 
     function factory() external view returns (address);
     function parentVault() external view returns (address);
     function asset() external view returns (address);
     function morpho() external view returns (address);
+    function marketIds(uint256 index) external view returns (bytes32);
+    function positions(bytes32 marketId) external view returns (uint128 supplyShares, uint128 allocation);
     function adapterId() external view returns (bytes32);
     function skimRecipient() external view returns (address);
-    function marketParamsList(uint256 index) external view returns (address, address, address, address, uint256);
-    function marketParamsListLength() external view returns (uint256);
-    function allocation(MarketParams memory marketParams) external view returns (uint256);
+    function marketIdsLength() external view returns (uint256);
+    function newAllocation(bytes32 marketId) external view returns (uint256);
```

Comment:
Still don't like newAllocation as an external function

Reply:
what's a better name

Reply:
realAssets is confusing because of a remark by @MathisGD:
> actually it's not incredible, because realAssets = 0 if the allocation is zero

So the scenario would be a market that was in the list before, was put to allocation 0 (but not shares 0) so removed from the list, and with interest accrual it increases in value and now has `realAssets > 0` but it doesn't contribute to the overall real assets